### ISSUE TITLE
Use v-prefixed tags in jquery-dist

### ISSUE THREAD

**Issue by bep** [jquery/jquery#4579]
I understand if you for some technical reason do not want/can do this, but I thought I'd ask:

In https://github.com/jquery/jquery-dist/releases you tag the version as `3.4.1`, which does not work for tools expecting the v-prefix (which seems to be a common practise). One example would be Hugo Modules (backed by Go Modules):

https://github.com/bep/docuapi/blob/977d645ab43c8541b9e2963cb56a06fbeb9a7540/go.mod#L6

Using the commit hash works, but it means that it gets slightly harder to maintain, it does not get to participate in the version selection...

**Comment by mgol** [comment]
Thanks for the report.

The tags we use (like `3.4.1`) _are_ semver versions - even more so than tags prefixed by `v` as those are not technically valid semver (see https://semver.org/ for the full grammar); it's just that the npm [semver package](http://npmjs.com/package/semver) often used in JS world to handle semver versions understands such a prefix and Node.js (via `node --version`) uses such a format (but npm itself - via `npm --version` is not, for example).

Therefore, I'd say that it's a tooling problem and we're doing the right thing here. Do you have any resources that would argument why the above reasoning is incorrect here?

**Comment by bep** [comment]
>Do you have any resources that would argument why the above reasoning is incorrect here?

The Semver mentions the "v" prefix:

>No, “v1.2.3” is not a semantic version. However, prefixing a semantic version with a “v” is a common way (in English) to indicate it is a version number. Abbreviating “version” as “v” is often seen with version control. Example: git tag v1.2.3 -m "Release version 1.2.3", in which case “v1.2.3” is a tag name and the semantic version is “1.2.3”.

Other than that I got nothing other than a hunch feeling that `v3.4.1` is much more common than `3.4.1` on GitHub. They even recommend it in the release page:

>It’s common practice to prefix your version names with the letter v. Some good tag names might be v1.0 or v2.3.4.

And in this particular case, I cannot do anything about the tooling (a more lenient strategy have been suggested by many, but rejected every time).

**Comment by mgol** [comment]
> The Semver doc mentions the "v" prefix:

Thanks, I missed that when scanning the document.

I understand the concerns and if we were setting up the infrastructure right now, your arguments could lead us to go the prefixed route. However:
1. The jQuery repository have used tags in the current format since always; way before semver itself was created. If we changed the tags now, who knows what tools that know how to work with our current repository structure might break.
1. We don't create tags by ourselves; it's a part of our [jquery-release project](https://github.com/jquery/jquery-release). This project is used by quite a few projects; changing that would require all of those projects to switch as well.
1. While the `v` prefix is allowed, it's not required by semver (correct me if I'm wrong). That means if a tool requires the `v` prefix on tags then it doesn't actually support semver as defined in https://semver.org/

In other words, I don't see a lot we can do here; onus is on the tools that claim they support semver IMO.

**Comment by bep** [comment]
>The jQuery repository have used tags in the current format since always; way before semver itself was created

I totally understand, and thanks for the quick response. I knew this was a long shot, but thought I'd give it a shot ...

**Comment by timmywil** [comment]
Thank you for the issue, but the team is in agreement that we won't be making a change here.
